worker-batch-ingestion-plugin/src/main/java/com/github/cafdataprocessing/worker/ingestion/validator/ValidationFileAdapter.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2019-2022 Micro Focus or one of its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.cafdataprocessing.worker.ingestion.validator;
+
+import org.json.JSONObject;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+public class ValidationFileAdapter
+{
+    private final JSONObject fieldsJsonObject;
+    private final JSONObject typesJsonObject;
+
+    public ValidationFileAdapter(final String file) throws ValidationFileAdapterException
+    {
+        final JSONObject fileContents = new JSONObject(getFileContents(file));
+        this.fieldsJsonObject = new JSONObject(fileContents.optString("fields"));
+        this.typesJsonObject = new JSONObject(fileContents.optString("types"));
+    }
+
+    public Set<String> getFieldKeys()
+    {
+        return fieldsJsonObject.keySet();
+    }
+
+    public Map<String, Set<String>> getFlattenedFieldKeys()
+    {
+        final Map<String, Set<String>> flattenedFields = new HashMap<>();
+        for (String fieldKey : getFieldKeys()) {
```

Comment:
Loop variables should also be `final`:

```suggestion
        for (final String fieldKey : getFieldKeys()) {
```